### ISSUE TITLE
[BUG] JSON.parse: expected ',' or '}' with long numeral comma-separated keyword string

### ISSUE THREAD

**Issue by nikAizuddin** [opensearch-project/OpenSearch-Dashboards#6017]
**Describe the bug**

Got `JSON.parse: expected ',' or '}' after property value in object at line 1 column 175 of the JSON data` when passing `"10080214256450617947,2031884987996059597,1"`
**To Reproduce**
1. Go to `Dev Tools`
2. Execute:
```json
POST myindex/_doc
{
  "@timestamp" : "2024-03-04T09:40:08.2531122Z",
  "JustAKeyword" : "10080214256450617947,2031884987996059597,1"
}
```
3. You'll see the `JSON.parse` error.

**Expected behavior**
There should be no error.

**OpenSearch Version**
Main branch commit https://github.com/opensearch-project/OpenSearch/tree/87ac37460c16a5b3cfa1cd85cad2cb7468b430a3

**Dashboards Version**
Main branch commit https://github.com/opensearch-project/OpenSearch-Dashboards/tree/9901bea341904421eb802a004476cdd313811b1e

**Plugins**
OpenSearch plugins:
```
opensearch-alerting
opensearch-asynchronous-search
opensearch-cross-cluster-replication
opensearch-custom-codecs
opensearch-flow-framework
opensearch-geospatial
opensearch-index-management
opensearch-job-scheduler
opensearch-knn
opensearch-ml
opensearch-neural-search
opensearch-notifications
opensearch-notifications-core
opensearch-observability
opensearch-performance-analyzer
opensearch-reports-scheduler
opensearch-security
opensearch-security-analytics
opensearch-skills
opensearch-sql
opensearch-time-series-analytics
```

Dashboard plugins:
```
alertingDashboards@3.0.0.0
anomalyDetectionDashboards@3.0.0.0
assistantDashboards@3.0.0.0
customImportMapDashboards@3.0.0.0
notificationsDashboards@3.0.0.0
observabilityDashboards@3.0.0.0
queryWorkbenchDashboards@3.0.0.0
reportsDashboards@3.0.0.0
searchRelevanceDashboards@3.0.0.0
ganttChartDashboards@3.0.0.0
indexManagementDashboards@3.0.0.0
mlCommonsDashboards@3.0.0.0
securityAnalyticsDashboards@3.0.0.0
securityDashboards@3.0.0.0
```

**Screenshots**

![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4245729/b82f5de5-89d7-48d1-b1fc-f4a82974d680)

**Host/Environment (please complete the following information):**
 - OS: Fedora 38
 - Browser and version:
     - Firefox 123.0 (64-bit)
     - Google Chrome 122.0.6261.94 (Official Build) (64-bit)
 - Node: v18.19.0
 - Java: Java(TM) SE Runtime Environment Oracle GraalVM 21.0.2+13.1 (build 21.0.2+13-LTS-jvmci-23.1-b30)

**Additional context**

All OpenSearch and Dashboard plugins are build from source

**Comment by MarkusAdvantage** [comment]
Have the same/similar issue but when searching data, extremely intermittent and cannot reproduce reliably. 
1. Searching index - Sysmon in this example - produce data as expected, refresh a few times works no issues. 
2. Search a minute later, error occurs. 
3. Give it a minute or two, search the same data frame and same data set, works again.
4. The time frame or data is irrelevant as I tested with dev tools, checked for valid json, various time frames. 
5. Sounds wishy-washy I know.

![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/153564476/20475fe2-81b8-43da-952e-894c90db9b61)

When the discover query is executed in dev tools:
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/153564476/cd21302e-b9d2-4265-a0a6-c619c51c2d01)

Tested getting data from the same index without the query parameters from discover and it produces data fine.

**OpenSearch Version**
v2.12.0 tar.gz

**Dashboards Version**
v2.12.0 tar.gz

**Plugins**
OpenSearch plugins:

opensearch-alerting
opensearch-anomaly-detection
opensearch-asynchronous-search
opensearch-cross-cluster-replication
opensearch-custom-codecs
opensearch-flow-framework
opensearch-geospatial
opensearch-index-management
opensearch-job-scheduler
opensearch-knn
opensearch-ml
opensearch-neural-search
opensearch-notifications
opensearch-notifications-core
opensearch-observability
opensearch-performance-analyzer
opensearch-reports-scheduler
opensearch-security
opensearch-security-analytics
opensearch-skills
opensearch-sql

**Dashboard plugins:**
alertingDashboards@2.12.0.0
anomalyDetectionDashboards@2.12.0.0
assistantDashboards@2.12.0.0
customImportMapDashboards@2.12.0.0
ganttChartDashboards@2.12.0.0
indexManagementDashboards@2.12.0.0
mlCommonsDashboards@2.12.0.0
notificationsDashboards@2.12.0.0
observabilityDashboards@2.12.0.0
queryWorkbenchDashboards@2.12.0.0
reportsDashboards@2.12.0.0
searchRelevanceDashboards@2.12.0.0
securityAnalyticsDashboards@2.12.0.0
securityDashboards@2.12.0.0

**Host/Environment**
OS: Ubuntu 22.04
Browser: Chrome version 122.0.6261.94 (Official Build) (64-bit)

**Comment by ananzh** [comment]
@AMoo-Miki could you comment on this? is this fixed?

**Comment by AMoo-Miki** [comment]
This was fixed with https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5956 which will be included in the 2.13.0 release.

**Comment by ashwin-pc** [comment]
This has been fixed in https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5956 and should be available in the next release.

**Comment by AMoo-Miki** [comment]
I have a version with more generalized rules but am having trouble testing them all.

@nikAizuddin using your sample query from the first post, I found a combination that allowed me to reproduce the issue.

@MarkusAdvantage intermittent problems indicate that your problem is something different; maybe you have a mix of version in your cluster?

**Comment by MarkusAdvantage** [comment]
@AMoo-Miki No, was the version indicated in my entry. I had to painfully downgrade to 2.11 and delete indices to get it working. In any case, will see if it is fixed with the new version, thank you.

**Comment by amirkkn** [comment]
> Have the same/similar issue but when searching data, extremely intermittent and cannot reproduce reliably.
> 
> 1. Searching index - Sysmon in this example - produce data as expected, refresh a few times works no issues.
> 2. Search a minute later, error occurs.
> 3. Give it a minute or two, search the same data frame and same data set, works again.
> 4. The time frame or data is irrelevant as I tested with dev tools, checked for valid json, various time frames.
> 5. Sounds wishy-washy I know.
> 
> ![image](https://private-user-images.githubusercontent.com/153564476/309886875-20475fe2-81b8-43da-952e-894c90db9b61.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2MDMxMzQsIm5iZiI6MTcxMjYwMjgzNCwicGF0aCI6Ii8xNTM1NjQ0NzYvMzA5ODg2ODc1LTIwNDc1ZmUyLTgxYjgtNDNkYS05NTJlLTg5NGM5MGRiOWI2MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOFQxOTAwMzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kODAyOGNmMjZkZTE0NDhlNWY1OWE0YTVjMDU2YTcwYzMxYTEwZTIzMjg3YzAzNjRlZmRiMjhjYTRhZDNlNWU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jGDV-N4tHGgN7sqbOkssqHU1tJ5x6TgbqqZhehkD8yg)
> 
> When the discover query is executed in dev tools: ![image](https://private-user-images.githubusercontent.com/153564476/309887825-cd21302e-b9d2-4265-a0a6-c619c51c2d01.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2MDMxMzQsIm5iZiI6MTcxMjYwMjgzNCwicGF0aCI6Ii8xNTM1NjQ0NzYvMzA5ODg3ODI1LWNkMjEzMDJlLWI5ZDItNDI2NS1hMGE2LWM2MTljNTFjMmQwMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOFQxOTAwMzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMzVjMjY5N2Q4NTUzMzQxZjM2MDQ5MmNlYWQ3YTM5MjU5NmM2ZjNhYTQzYWY0ODg1M2EyN2IzYjc0YTZiZjY0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.JkCf-88iTj6vnW72H5Mo1qtuy23wIjJxXkUETkReyOo)
> 
> Tested getting data from the same index without the query parameters from discover and it produces data fine.
> 
> **OpenSearch Version** v2.12.0 tar.gz
> 
> **Dashboards Version** v2.12.0 tar.gz
> 
> **Plugins** OpenSearch plugins:
> 
> opensearch-alerting opensearch-anomaly-detection opensearch-asynchronous-search opensearch-cross-cluster-replication opensearch-custom-codecs opensearch-flow-framework opensearch-geospatial opensearch-index-management opensearch-job-scheduler opensearch-knn opensearch-ml opensearch-neural-search opensearch-notifications opensearch-notifications-core opensearch-observability opensearch-performance-analyzer opensearch-reports-scheduler opensearch-security opensearch-security-analytics opensearch-skills opensearch-sql
> 
> **Dashboard plugins:** alertingDashboards@2.12.0.0 anomalyDetectionDashboards@2.12.0.0 assistantDashboards@2.12.0.0 customImportMapDashboards@2.12.0.0 ganttChartDashboards@2.12.0.0 indexManagementDashboards@2.12.0.0 mlCommonsDashboards@2.12.0.0 notificationsDashboards@2.12.0.0 observabilityDashboards@2.12.0.0 queryWorkbenchDashboards@2.12.0.0 reportsDashboards@2.12.0.0 searchRelevanceDashboards@2.12.0.0 securityAnalyticsDashboards@2.12.0.0 securityDashboards@2.12.0.0
> 
> **Host/Environment** OS: Ubuntu 22.04 Browser: Chrome version 122.0.6261.94 (Official Build) (64-bit)

was the issue solved after upgrade to 2.13.0? We have the same issue after upgrade to 2.12.0

**Comment by ashwin-pc** [comment]
@amirkkn yes this was fixed in the 2.13 release
